src/widget-sources/widget-base-source.ts

```diff
@@ -93,14 +93,26 @@ export abstract class WidgetBaseSource<Props extends WidgetBaseSourceProps> {
   async getCategories(
     options: CategoryRequestOptions
   ): Promise<CategoryResponse> {
-    const {filterOwner, spatialFilter, abortController, ...params} = options;
+    const {
+      filterOwner,
+      spatialFilter,
+      spatialFiltersMode,
+      spatialFiltersResolution,
```

Comment:
I'm not sure yet if we _should_ do this, so no need to make the change now, but ... would it work if instead of asking the user to provide `spatialFiltersResolution`, we computed it internally from the WidgetBaseSource's own props, and a (new) viewState prop that the user would pass into `getCategories` etc.? Or is this something the user needs to directly control?

I'm a bit worried about all of the default values used inside 'getSpatialFiltersResolution', and things like `tileSize`, because it would be easy for the user to forget something. It would be nice if all the default values were populated before the 'source' is passed into the `getSpatialFiltersResolution` function, and this would make implementation in the widget web components simpler.

Reply:
Yes, user should be able to specify a concrete `spatialFiltersResolution` I think. I'm also a bit worried about all the conditional logic inside of that function, but since is inherited legacy code, I decided not to question it very much. We can find a way to adapt it if you would like.

Reply:
/cc @zbigg @felixpalmer does this concern make sense, and do you have any immediate preferences? 

If we're all not sure then maybe it's best to go ahead and try setting up 1-2 of the example widgets (https://github.com/CartoDB/carto-api-client/tree/main/examples/components/widgets) to test with spatial index sources.

Reply:
Bear in mind I am not familiar with this code, so feel free to disregard the following :)

Am I understanding correctly that by providing `spatialFiltersResolution` the user will be able to override what `getSpatialFiltersResolution` would otherwise provide?

`getSpatialFiltersResolution` is a complex function and especially with h3 there has been a lot of confusion about what "resolution" means and how it changes with the viewport.

As such I would vote for not exposing this parameter directly, but instead let the user specify a `spatialResolutionBias`, i.e. a value which increases/decreases the resolution that would otherwise be used. So passing `2` would give double the resolution compared to the default.

Reply:
@felixpalmer I agree `getSpatialFiltersResolution` is a complex function and does not need to be exported

We have added more options with merging [this PR](https://github.com/CartoDB/carto-api-client/pull/35) here, where you pass a viewState (latitude, longitude and zoom) param instead of calling the function and the function gets called internally with that. But still we are allowing to receive a `spatialFiltersResolution` param

Reply:
And for the `spatialResolutionBias`, I can see how it would be applied to h3 resolution because there is a BIAS constant there, but how would we apply it to quadbins? Makes sense to apply to one and not the other?

Reply:
I don't think there is any use case (CARTO-internal or external) that currently requires either `spatialFiltersResolution` or `spatialResolutionBias`, correct? If not then perhaps we should just wait on exposing it.

Reply:
Ok, using this commit https://github.com/CartoDB/carto-api-client/pull/34/commits/f85f818305e41218c9c2e597ee74e852c0977e05 to not export the `getSpatialFiltersResolution` function